### ISSUE TITLE
[SF][SelectDialog]: [A11y][Selected list items count visually updated above list but is not read out by screen reader]

### ISSUE THREAD

**Issue by CharliDong** [SAP/ui5-webcomponents-react#7592]
### Describe the bug

In UI5, when using the Select Dialog in multiple mode, after selecting an item from the list, the screen reader announces: "selected item: 1."  his announcement is not made in the webcomponent.

### Reproduction steps

1. Login using URL and instance.
    https://qacand.qa.hr.cloud.sap/login
    CDC04H/cgrant/pwd
    In case password was changed, try pwd/pwd1
2. Navigate to Goals from module switcher -> Click Development Goals Tab and Click Create Goals.
3. Click on Create from Roles and Create from Interests buttons.
4. Turn ON screen reader. Click any item.

### Expected Behaviour

The screen reader should announces: "selected item: 1." 

### UI5 Web Components for React Version

2.10.0

### UI5 Web Components Version

2.10.0

### Browser

Chrome

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.
